raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -287,4 +287,7 @@ public String name() {
         return "Leader";
     }
 
+    @Override
+    public void close() {}
```

Comment:
Maybe we could add a default no-op implementation to EpochState?

Reply:
I prefer to have each of the epoch states explicitly opt out of this `close` method. This makes it clear that this state doesn't have any resource that it wishes to clean/close during a transition. Instead of future code changes forgetting to override this method. What do you think?